essentials-tp/src/main/kotlin/io/github/nyliummc/essentials/util/TPCache.kt

```diff
@@ -0,0 +1,21 @@
+package io.github.nyliummc.essentials.util
+
+import com.mojang.authlib.GameProfile
+import net.minecraft.util.math.Vec3i
+
+/*
+ * Keeps track of last location teleported from for /back
+ */
+object TPCache {
+    private val cache: MutableMap<GameProfile, Pair<Vec3i, Int>> = mutableMapOf()
```

Comment:
I'd say use UUIDs to keys here, usernames aren't nessecary here and other junk like skin textures live on the profile.

Reply:
Also a Pair seems like a pretty bad thing to store. Maybe a sort of `Entry` class?